docs/product-business-resource-planner-guide/src/main/asciidoc/optashift-ER-openshift-deploy-proc.adoc

```diff
@@ -0,0 +1,105 @@
+.Procedure
+. Log in to the Openshift web console.
+. Open the project to host the starter application.
+. Select *Add to Project > Import YAML / JSON*.
+. In the Import YAML / JSON dialog box:
+.. Click the Browse button and navigate to the optashift-employee-rostering-template.yaml file. If you are starting from the optashift-employee-rostering folder, access the file by clicking *sources > openshift > templates*.
+.. Click Create.
++
+.Upload template window
+image::optashift-rostering/optashift-ER-s2i-step-one.png[]
++
+. In the Add Template dialog box, accept the default settings by clicking Continue.
++
+.Add template window
+image::optashift-rostering/optashift-ER-s2i-step-two.png[]
++
+. In the Import YAML/JSON dialog box, accept the default settings by clicking Create.
++
+.Configure template window
+image::optashift-rostering/optashift-ER-s2i-step-three.png[]
++
+. In the Success dialog box, click Close.
+. Allow the build and deployment to complete. This may take several minutes. When the deployment completes, click the resulting link to open the application.
++
+.Completed deployment and resulting link
+image::optashift-rostering/optashift-ER-s2i-step-four.png[]
+
+== Deploy with pre-built binary
+
+.Procedure
+. Prepare the binary deployment.
++
+[NOTE]
+====
+You must conduct binary deployments using the oc command line tools.  See the OpenShift documentation for installation steps.
+====
++
+.. Working from the optashift-employee-rostering-{VERSION} distribution folder, open the Binaries folder.
+.. Copy the optashift-employee-rostering-webapp-{VERSION}.war.
+.. Move the copied file to the Sources folder and rename it as ROOT.war. The updated name instructs the application to deploy the file to the root context of the server.
++
+. Create a project using the following code snippet:
++
+[source]
+----
+oc new-project optashift-demo --display-name="Employee Rostering Demo"
+----
++
+. Create a build and deployment configuration. Use a template file (optashift-employee-rostering-template-binary.yaml) to stub out the Openshift configuration, such as the route to expose the application to make it accessible to browsers.
++
+[source]
+----
+oc process -f openshift/templates/optashift-employee-rostering-template-binary.yaml -n optashift-demo | oc create -f - -n optashift-demo
+----
++
+[NOTE]
+====
+If you are deploying to an existing project, replace all instances of "optashift-demo" with your project name.
+====
++
+. Upload the binary file to begin a deployment.
++
+[source]
+----
+oc start-build employee-rostering --from-file=ROOT.war
+----
++
+. View the deployed application details. Click the link in the Routes section to open the starter application.
++
+.Openshift web console with deployed Employee Rostering starter application
+image::optashift-rostering/optashift-ER-binary-success.png[]
++
+
+//.Procedure
+//. Build from local source
+// ./provision.sh setup employee-rostering --binary
+// This doesn't work in current release
```

Comment:
Is there a jira for this? can we reference the jira url in this comment and add a TODO?

Reply:
Isn't this a reference to the BXMSPROD-1 issue where a build from source simply does not work at present and Ant is saying it's gonna be fixed soon and it isn't getting fixed yet?

Reply:
There is a comment including a link to the JIRA if you view the raw source